### PR TITLE
JWT secret rotation

### PR DESCRIPTION
**Pull request by graetzer**
https://github.com/arangodb/arangodb/compare/feature/live-jwt-secret-change?expand=1

### CODE REVIEW

File: 3.7/http/general.md [hunk 1]
```diff
@@ -242,6 +242,53 @@ jwtgen -s <my-secret> -e 3600 -v -a "HS256" -c 'iss=arangodb' -c 'server_id=mycl
 curl -v -H "Authorization: bearer $(jwtgen -s <my-secret> -e 3600 -a "HS256" -c 'iss=arangodb' -c 'server_id=myclient')" http://<database-ip>:8529/_api/version
 ```
 
+#### Hot-Reload of JWT Secrets
+
+To perform a hot reload the JWT secrets of a local arangod process you may use the following 
+HTTP restful API. Perform a **POST** request to reload the secret, a **GET** request may 
+be used to load information obout the currently used secrets.
```
Comment (informalict) [inline 3.7/http/general.md]: typo: obout -> about

File: 3.7/http/general.md [hunk 2]
```diff
@@ -242,6 +242,57 @@ jwtgen -s <my-secret> -e 3600 -v -a "HS256" -c 'iss=arangodb' -c 'server_id=mycl
...
+Hot-Reload the secrets from disk:
+
+`POST /_admin/server/jwt`
+
+To fetch information about the currently loaded secrets:
+
+`GET /_admin/server/jwt`
+
+A successful response contains a JSON with the "result" field and an "error" field containing _false_.
+The field _result_ which in turn has the fields _active_ and _passive_, which contain the sha256 hashes of the loaded secrets.
```
Comment (jsteemann) [inline 3.7/http/general.md]:
```suggestion
The field _result_ contains the sub-fields _active_ and _passive_, which contain the sha256 hashes of the loaded secrets.
```

File: 3.7/http/general.md [hunk 3]
```diff
@@ -242,6 +242,57 @@ jwtgen -s <my-secret> -e 3600 -v -a "HS256" -c 'iss=arangodb' -c 'server_id=mycl
...
+The field _passive_ is always an array.
```
Comment (jsteemann) [inline 3.7/http/general.md]:
```suggestion
The _field_ active is always an object, the field _passive_ is always an array of objects. However, _passive_ may be an empty array if there is only an active secret but no passive ones.
```

File: 3.7/programs-arangod-server.md [hunk 4]
```diff
@@ -191,20 +191,45 @@ operating systems do not provide this option and will ignore it.
...
-generate a secret.
+In single server setups ArangoDB will generate a secret if none was specified.
 
 In cluster deployments which have authentication enabled a secret must
 be set consistently across all cluster nodes so they can talk to each other.
 
+ArangoDB also supports an option to pass the secret without a file, however
+this is discouraged for security reasons.
+
+`--server.jwt-secret >secret>`
```
Comment (jsteemann) [inline 3.7/programs-arangod-server.md]:
```suggestion
`--server.jwt-secret <secret>`
```

File: 3.7/programs-arangod-server.md [hunk 5]
```diff
@@ -191,20 +191,45 @@ operating systems do not provide this option and will ignore it.
...
+
+You may use multiple secrets, where the _active_ secret is used to sign new JWT tokens
+and all other _passive_ secrets are just used to validate incoming JWT tokens.
+
+
+`--server.jwt-secret-folder <folder-with-secrets>`
+
+The list of files in this folder is sorted alphabetically. The first is used
+as the _active_ secret to sign new tokens. All other secrets are passively
+used during verification. Only one secret needs to verifiy a JWT token
```
Comment (jsteemann) [inline 3.7/programs-arangod-server.md]:
```suggestion
used during verification. Only one secret needs to verify a JWT token
```

**Comment by jsteemann** [comment]
This new feature should definitely be mentioned in the file `3.7/release-notes-new-features37.md` as a new enterprise feature!

**Pull request merged**